modeshape-jcr/src/main/java/org/modeshape/jcr/cache/document/BucketId.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.jcr.cache.document;
+
+import java.util.Objects;
+import org.modeshape.common.util.SecureHash;
+import org.modeshape.jcr.value.Name;
+
+/**
+ * Abstraction for the ID of a bucket.
+ * 
+ * @author Horia Chiorean (hchiorea@redhat.com)
+ * @see BucketedChildReferences
+ */
+class BucketId implements Comparable<BucketId> {
+    private final String id;
+
+    protected BucketId( Name name, int prb ) {
+        this(name.getString(), prb);
+    }
+
+    protected BucketId( String nameString, int prb ) {
```

Comment:
Obviously `prb` used to determine the length of the SHA1 that is used, but I can't figure out what `prb` stands for.